src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -271,6 +278,31 @@ def lower_to_language_level(self):
         call = Call.create(routine_symbol, arguments)
         self.replace_with(call)
 
+        # Remove original invoke symbol if there are no other
+        # references to it. This is not strictly necessary but it
+        # tidies things up and can avoid an exception being raised in
+        # the Fortran Writer as the invoke symbol has an
+        # UnresolvedInterface.
+
+        first = True
```

Comment:
I think this would be nicer as:

    symbol_table = call.scope.symbol_table
    while symbol_table:
        try:
            invoke_symbol = symbol_table.lookup("invoke", scope_limit=symbol_table.node)
        except KeyError:
            symbol_table = symbol_table.parent_symbol_table()
            continue
        if not symbol_table.node.walk(AlgorithmInvokeCall):
            symbol_table.remove(invoke_symbol)
        break
    else:
        ...

Reply:
Yes that is much better.